lib/tools/settings-client-commands.js

```diff
@@ -39,8 +39,26 @@ const GPS_CACHE_REFRESHED_LOGS = [
 
 const GPS_COORDINATES_PATTERN = /data="(-?[\d.]+)\s+(-?[\d.]+)\s+(-?[\d.]+)"/;
 
+const FORE_GROUND_APP_KEYWORD = 'isForeground=true';
+
 const commands = {};
 
+/**
+ * If the io.appium.settings package has running foreground service.
+ *
+ * @this {import('../adb.js').ADB}
+ * @returns {Promise<boolean>} Return true if the device has running settings app foreground service.
+ */
+commands.hasRunningSettingsAppForegroundService = async function hasRunningSettingsAppForegroundService () {
```

Comment:
note: api level 28+ worked with the `Active foreground apps` (this way), but at least Android 7 seems not for the case. The main issue I found by `cmd notification` was api level 29+, so this method can follow the old pid existing way for api level 27 and lower I think

Reply:
(updated. So api level 25 and lower can keep the pid way)